src/parsers/InputFieldDeclarationParser.ts

```diff
@@ -56,7 +85,50 @@ export class InputFieldDeclarationParser {
 
 	static templates: Template[] = [];
 
-	static parse(fullDeclaration: string): InputFieldDeclaration {
+	static parseDeclaration(
+		fullDeclaration: InputFieldDeclaration,
+		inputFieldArguments: Record<InputFieldArgumentType, string> | {} | undefined | null = undefined,
+		templateName: string | undefined | null = undefined
+	): InputFieldDeclaration {
+		// field type check
+		fullDeclaration.inputFieldType = InputFieldDeclarationParser.getInputFieldType(fullDeclaration.inputFieldType);
+
+		// template check:
+		let useTemplate: boolean = isTruthy(templateName) && typeof templateName === 'string';
```

Comment:
i don't think that the typeof check is necessary here, since all the cases should already be covered by the `isTruthy` check